service/migration_manager.cc

```diff
@@ -746,7 +748,25 @@ future<std::vector<mutation>> prepare_column_family_update_announcement(storage_
         }
         co_await seastar::async([&] {
             db.get_notifier().before_update_column_family(*cfm, *old_schema, mutations, ts);
+
+            // Notify drop indices
+            auto diff = difference(old_schema->all_indices(), cfm->all_indices());
```

Comment:
This duplicates logic from make_update_indices_mutations() which is called from make_update_table_mutations(), invoked above. Why not call the notifier there?

Reply:
First, sending notifications from schema_tables seems like a layering violation.
Second, this is not an exact duplicate.
make_update_indices_mutations calls `cf.get_index_manager().create_view_for_index` which creates a new view schema, while for this purpose we need `db.find_schema` since the index's tablets use that table_id.
We can also look up the view schema in the table's `views()`, but that's a bit less efficient.

Reply:
> First, sending notifications from schema_tables seems like a layering violation.

We already call db.get_notifier().before_create_column_family() from there. Note that this is not a generic notification, but an extension point for mutation building. It's not supposed to change the db state, just add more mutations to the result.

> view schema, while for this purpose we need db.find_schema since the index's tablets use that table_id.

We can call find_schema() from there.